server/bin/import_test_data.rb

```diff
@@ -207,7 +207,8 @@ def create_product(cp, owner, product)
     :attributes => attrs,
     :dependentProductIds => dependent_products,
     :relies_on => relies_on,
-    :branding => branding
+    :branding => branding,
+    :providedProducts => provided_products
```

Comment:
1. Products have provided products. When we create product as derived products, provided products property will behave like derived provided products for it. There is no need to add derived provided product explicitly into product.
2. I thought, for create_mkt_product_and_pools, code of adding provided products into a pool will be removed under ENT-1832.

Reply:
Regarding 2: This file (import_test_data.rb), is only called by the deploy script to load the test data codified in test_data.json, so it is unrelated to the rest of the changes or the rspec tests, and doesn't need to happen as part of this task. (this test data do not affect rspec tests at all, and are only used for manual testing after we run deploy script with the -t flag). This is what ENT-1834 was created for. So there is no need for it to be changed incrementally, and it will be cleaner to do it all at once, either in this PR, or in a separate PR specifically for ENT-1834.

About 1: You are right that there is no derived provided products field on the Product model any more, but in the test_data.json, we still codify it like that (there is a single test product with derived_product_id / derived_provided_products fields). So here, instead of always doing `:providedProducts => provided_products`, we should add some logic that looks like this:

If derived_product_id is there, then do this: `:providedProducts => derived_provided_products`
otherwise, do this: `:providedProducts => provided_products`

Reply:
Regarding 2: I needed to change the method `def create_product(cp, owner, product)` of `import_test_data.json` beacuse it's used in various rspec test methods. I'll update the necessary changes required for card ENT-1834 in this PR only.

About 1: As per my understanding, in test_data.json, when derived product JSON is created,  it's equivalent derived provided products should also be a part of it. And we should be able to create the derived product with it's derived provided products.

Reply:
Removed the code of  passing provided products from `create_product` method.